From 612dab541486a0f4732b4ab97245cecb3df1f2cc Mon Sep 17 00:00:00 2001
From: leskal <alexakis1997@gmail.com>
Date: Mon, 5 Sep 2016 12:33:47 +0300
Subject: [PATCH] mtk core

Change-Id: Id64e2366ae957aa3032b5f8e160c2089ae13db5e

diff --git a/healthd/BatteryMonitor.cpp b/healthd/BatteryMonitor.cpp
index 86a4fc6..584a466 100644
--- a/healthd/BatteryMonitor.cpp
+++ b/healthd/BatteryMonitor.cpp
@@ -207,7 +207,11 @@ bool BatteryMonitor::update(void) {
     props.batteryLevel = mBatteryFixedCapacity ?
         mBatteryFixedCapacity :
         getIntField(mHealthdConfig->batteryCapacityPath);
+#ifndef MTK_HARDWARE
     props.batteryVoltage = getIntField(mHealthdConfig->batteryVoltagePath) / 1000;
+#else
+    props.batteryVoltage = getIntField(mHealthdConfig->batteryVoltagePath);
+#endif
 
     props.batteryTemperature = mBatteryFixedTemperature ?
         mBatteryFixedTemperature :
diff --git a/init/init.cpp b/init/init.cpp
index 58d7d34..450fec2 100644
--- a/init/init.cpp
+++ b/init/init.cpp
@@ -85,7 +85,9 @@ static int have_console;
 static char console_name[PROP_VALUE_MAX] = "/dev/console";
 static time_t process_needs_restart;
 
-static const char *ENV[32];
+// some MTK services (e.g. ril-daemon-mtk) require very large number
+// of sockets, which can't be contained in 32 entries minus other variables.
+static const char *ENV[64];
 
 bool waiting_for_exec = false;
